### ISSUE TITLE
Regression many to many

### ISSUE THREAD

**Issue by joelrich** [philipperemy/keras-tcn#7]
I have tried to use your library to do a many to many regression, which does not seem to be supported out of the box. Would you mind to indicate what has to be changed for this use case?

**Comment by philipperemy** [comment]
@joelrich in your case, you should probably use a TimeDistributed layer to make inference for more than one step ahead. Assuming you're predicting sequences.

https://keras.io/layers/wrappers/

If you're just predicting independent variables, then a Dense(nb_classes) without a softmax would also work I guess.

**Comment by philipperemy** [comment]
(Feel free to reopen it if it's unclear)

**Comment by joelrich** [comment]
@philipperemy thank you. I got it running by adjusting `Dense(1)` in the following lines:
```
    else:
        # regression
        x = Dense(1)(x)
        x = Activation('linear', name='output_dense')(x)
```

**Comment by philipperemy** [comment]
@joelrich ok that will work in all the cases. You have to keep in mind that it won't be optimal if you have a time-dependency in what you predict.

```
 Dense(4)(x) => will predict 4 independent values at the same time.
It's like predicting P(X1, X2, X3, X4 | Z).
```

If you have a sequence, it's usually better to model and consider this likelihood:
```
P(X1 | Z)
+ P(X2 | X1, Z)
+ P(X3 | X1, X2, Z)
+ P(X4 | X1, X2, X3, Z)
```

**Comment by joelrich** [comment]
@philipperemy currently, I am predicting independent values. Thanks for the advice anyway (might be relevant later on).

**Comment by swilson314** [comment]
@philipperemy I can't decide whether I'm being really dense or this is a nomenclature issue (or both). If I set regression=True and return_sequences=True, isn't this many to many regression?

**Comment by philipperemy** [comment]
Regression=True => regression task with output = 1
Regression=False => classification task

regression=True and return_sequences=True won't give you a many to many regression.

Many to many is actually not handled properly in the compile_tcn code. 

https://github.com/philipperemy/keras-tcn/blob/6d71e1946910748a4fb99ce53b29d312ddfb82b2/tcn/tcn.py#L250

You have to update manually this line.

I'd like to work on it but I'm a bit busy at the moment...

**Comment by philipperemy** [comment]
@swilson314 yes what you're saying totally makes sense. I'll try to find some time soon to implement this.

**Comment by swilson314** [comment]
@philipperemy Haha, my point is I think many to many is already implemented. I'm putting together an example to verify.

**Comment by philipperemy** [comment]
@swilson314 ok please go ahead ;)

**Comment by swilson314** [comment]
@philipperemy I didn't even need to create a new example. I just took the copy_memory example and changed classes=0 and regression=True -- this is a regression problem that has many input time steps and many output time steps.

So I think this is just a nomenclature issue. Per https://machinelearningmastery.com/encoder-decoder-long-short-term-memory-networks/

> Given that there are multiple input time steps and multiple output time steps, this form of problem is referred to as many-to-many type sequence prediction problem.
